doc/sphinx/language/core/records.rst

```diff
@@ -3,208 +3,313 @@
-   .. insertprodn record_definition field_def
+   .. insertprodn record_definition field_spec
 
    .. prodn::
       record_definition ::= {? > } @ident_decl {* @binder } {? : @sort } {? := {? @ident } %{ {*; @record_field } {? ; } %} {? as @ident } }
-      record_field ::= {* #[ {*, @attribute } ] } @name {? @field_body } {? %| @natural } {? @decl_notations }
-      field_body ::= {* @binder } @of_type
-      | {* @binder } @of_type := @term
+      record_field ::= {* #[ {*, @attribute } ] } @name {? @field_spec } {? %| @natural }
+      field_spec ::= {* @binder } @of_type
       | {* @binder } := @term
-      term_record ::= %{%| {*; @field_def } {? ; } %|%}
-      field_def ::= @qualid {* @binder } := @term
-
-   Each :n:`@record_definition` defines a record named by :n:`@ident_decl`.
-   The constructor name is given by :n:`@ident`.
-   If the constructor name is not specified, then the default name :n:`Build_@ident` is used,
-   where :n:`@ident` is the record name. If given, the :n:`as @ident`
-   part specifies the name to use for inhabitants of the record in the
-   type of projections (see below).
-
-   If :token:`sort` is omitted, the default sort is Type.
-   Notice that the type of an identifier can depend on a previously-given identifier. Thus the
-   order of the fields is important. :n:`@binder` parameters may be applied to the record as a whole
-   or to individual fields.
-
-   .. todo
-      "Record foo2:Prop := { a }." gives the error "Cannot infer this placeholder of type "Type",
-      while "Record foo2:Prop := { a:Type }." gives the output "foo2 is defined.
-      a cannot be defined because it is informative and foo2 is not."
-      Your thoughts?
+      | {* @binder } @of_type := @term
+
+   Defines a non-recursive record type, creating projections for each field
+   that has a name other than `_`. The field body and type can depend on previous
+   fields, so the order of fields in the definition may matter.
+
+   Use the :cmd:`Inductive` and :cmd:`CoInductive` commands to define recursive
+   (inductive or coinductive) records.  These commands also permit defining
+   mutually recursive records provided that all of
+   the types in the block are records.  These commands automatically generate
+   induction schemes.  Enable the :flag:`Nonrecursive Elimination Schemes` flag
+   to enable automatic generation of elimination schemes for :cmd:`Record`.
+   See :ref:`proofschemes-induction-principles`.
 
    :n:`{? > }`
-     If provided, the constructor name is automatically declared as
+     If specified, the constructor is declared as
      a coercion from the class of the last field type to the record name
      (this may fail if the uniform inheritance condition is not
-     satisfied).  See :ref:`coercions`.
+     satisfied). See :ref:`coercions`.
+
+   :n:`@ident_decl`
+     The :n:`@ident` within is the record name.
+
+   :n:`{* @binder }`
+     :n:`@binder`\s may be used to declare the *parameters* of the record.
```

Comment:
"parameters" is too general a term.  Perhaps "record parameters".  I thought this wording was a bit vague but still understandable.  Seemed like it might take a while to understand it well enough to explain clearly.

Reply:
My question was precisely whether "record parameters" should be interpreted in the same way as "inductive parameters".

Reply:
@herbelin, do you know (and can suggest wording)?

Reply:
How about saying `the (inductive) *parameters* of the record`?

Reply:
I wouldn't add "inductive" unless I understand that it's different from "record parameters".  (I don't change things I know I don't understand.)

Reply:
But to answer @Zimmi48's question, parameters for records are the same for inductive types (which records happen to be a special case of).

Reply:
Thanks, then I reiterate my suggestion as such:
```suggestion
     :n:`@binder`\s may be used to declare the :term:`parameters` of the record.
```

Where "parameters" would be defined in the inductive chapter. But this is just a suggestion, and could also be deferred.

Reply:
Got it.  I made it "inductive parameters".